tests/model_serving/model_server/inference_graph/conftest.py

```diff
@@ -0,0 +1,90 @@
+import time
+from typing import Generator, Any
+
+import pytest
+from _pytest.fixtures import FixtureRequest
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.inference_graph import InferenceGraph
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.namespace import Namespace
+from ocp_resources.secret import Secret
+from ocp_resources.serving_runtime import ServingRuntime
+
+from utilities.constants import ModelFormat, KServeDeploymentType, ModelStoragePath
+from utilities.inference_utils import create_isvc
+
+
+@pytest.fixture
+def dog_breed_inference_graph(
+    admin_client: DynamicClient,
+    model_namespace: Namespace,
+    dog_cat_inference_service: InferenceService,
+    dog_breed_inference_service: InferenceService,
+) -> Generator[InferenceGraph, Any, Any]:
+    nodes = {
+        "root": {
+            "routerType": "Sequence",
+            "steps": [
+                {"name": "dog-cat-classifier", "serviceName": dog_cat_inference_service.name},
+                {
+                    "name": "dog-breed-classifier",
+                    "serviceName": dog_breed_inference_service.name,
+                    "data": "$request",
+                    "condition": "[@this].#(outputs.0.data.1>=0)",
+                },
+            ],
+        }
+    }
+    with InferenceGraph(
+        client=admin_client,
+        name="dog-breed-pipeline",
+        namespace=model_namespace.name,
+        nodes=nodes,
+    ) as inference_graph:
+        inference_graph.wait_for_condition(inference_graph.Condition.READY, "True")
+        time.sleep(5)  # This wait helps OpenShift to fully enable the Route, if the InferenceGraph is exposed
```

Comment:
iiuc the wait here for a route to be created? if so, please add wait for route using TimeoutSampler
using `sleep` is risky as different envs behave differently; if the route is not ready in expected time, this should fail.

Reply:
IIRC, the `Route` is created quickly, even before the previous line `wait_for_condition` would return.

The problem that I found was that after the IG is flagged as `Ready`, immediately sending a request to the IG endpoint results in the _Application not available_ page being returned. This small sleep is allowing the Route to settle down. 

Maybe, a better approach can be to probe the endpoint until the  _Application not available_ page is no longer returned???

Reply:
we have a retry on `run_inference` but saw yesterday that sometimes the response is not an err (i.e in res), so i will work on fixing it.
better to have the check in one place, so the sleep here should be removed

Reply:
Oh, I now see the `@retry` decorator on `run_inference`.

> but saw yesterday that sometimes the response is not an err (i.e in res)

I think the _Application is not available_ page would fall on this case. I tried curl on a dummy Route and the exit status of the `curl` command is zero. So, it won' be catched in `res`.

> better to have the check in one place, so the sleep here should be removed

OK, I support this.
I've removed the sleep and and modified run_inference to retry if the route is not ready yet. I hope the changes are acceptable.